### PR TITLE
Feature friendlier segmented control

### PR DESCRIPTION
**Pull request by livecodeali**
Various improvements to the segmented control
- refactored selection related code so it is more robust
- documented the messages it sends
- tweaked various old default properties
- updated the existing property docs, moved inline, moved metadata to follow property it relates to
- added a segmentCount property to easily add or remove segments.
- use this https://github.com/livecode/livecode-ide/pull/501 to add a multi-icon editor for the icon properties (defaults to current string editor until that PR is merged)

### CODE REVIEW

File: extensions/widgets/segmented/segmented.lcb [hunk 1]
```diff
@@ -541,10 +644,30 @@ private handler updateProperties() returns nothing
...
+	else if (the number of elements in mSegmentNames) > mNumSegments then
+		repeat with tX from 1 up to (the number of elements in mSegmentNames)-mNumSegments
+			delete element (tX + mNumSegments) of mSegmentNames
+		end repeat
+	end if
 		
 	if (the number of elements in mSegmentIcons) < mNumSegments then
 		repeat with tX from 1 up to (mNumSegments-(the number of elements in mSegmentIcons))
-			push "f10c" onto back of mSegmentIcons
+			push "circle" onto back of mSegmentIcons
```
Comment (peter-b) [inline extensions/widgets/segmented/segmented.lcb]: What about adding a `kDefaultSegmentIcon` constant?

Comment (livecodeali) [inline extensions/widgets/segmented/segmented.lcb]: good idea

**Comment by peter-b** [comment]
@livecode-vulcan review ok 2ac4f0d

**Comment by livecode-vulcan** [comment]
💙 review by @peter-b ok 2ac4f0d4c63c4f39484f199fedb48f8725deded8

**Comment by livecode-vulcan** [comment]
😎 test success 2ac4f0d4c63c4f39484f199fedb48f8725deded8
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/472)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/475)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/471)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/471)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/479)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/472)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/481)

**Pull request merged**